### ISSUE TITLE
Features

### ISSUE THREAD

**Issue by gnmmarechal** [chaoskagami/corbenik#55]
"It has a few features that no other CFW has:
IPC logging (which logs commands from/to services).
Arbitrary memory read/write over the network.
Testing services calls over the network.
Some kind of debugging with breakpoints over the network? Not sure what I'm reading in 3dshaxclient.c.
The memory setup for webkitdebug.s from https://github.com/yellows8/3ds_browserhax_common.
Some nim URL patches (though that effect can also be achieved, for the most part, with ctr-httpwn + another CFW)."
~beefhash, /r/3dshacks

I read that. Is any of these useful to be implemented in Corbenik, given that it is a poweruser/dev-oriented CFW?

**Comment by chaoskagami** [comment]
I assume you're referring to [3dshax](https://github.com/yellows8/3dshax). I guess I'll go down the list one by one:

IPC logging is...complicated. I personally think that would be *very* useful, but that's annoyingly complicated. I looked into it a while back. I still would like to implement this someday.

3dshaxclient: No. Let's just say I'd far prefer a gdb stub. I'm also not a fan of opening unauthenticated remote holes. While our consoles may be hacked, they shouldn't be insecure.

webkitdebug.s: Doable.

NIM url patches: ctr-httpwn handles more cases and selectively redirects only what is needed to get online. Replacing the NIM URL is doable in the same fashion as mysterymachine if that's really desired. Honestly, ctr-httpwn is a better solution. From reading 3dshax, I suspect this part was implemented before he wrote ctr-httpwn.

---

Since we're talking about 3dshax, I should also probably point out that yellows8 has not given a license for that project (or any, for that matter.) Minor patches to Nintendo's code are probably non-copyrightable. However, I can't legally include any actual code from his repo since that would be mixing GPL with unlicensed code.

Closing, since this is mostly a feature request and I'm not planning to act on it right now.
